pkg/watermark/fetch/interface.go

```diff
@@ -30,4 +30,6 @@ type Fetcher interface {
 	GetWatermark(offset isb.Offset) processor.Watermark
 	// GetHeadWatermark returns the latest watermark based on the head offset
 	GetHeadWatermark() processor.Watermark
+	// GetPodWatermarks returns the watermarks for all the pods in a vertex
+	GetPodWatermarks() []processor.Watermark
```

Comment:
I think this can be derived from other functions. Does it make sense to move it under UI since only UI is using it?